### ISSUE TITLE
How does weights work in calculating loss?

### ISSUE THREAD

**Issue by DeanLeeFumu** [catboost/catboost#1470]
Problem: 

Setting weights in Pool object often leads to larger loss compare to Non-weighted training.
I'm curious how to replicate the same calculation out side of training loop.

For example: 
Non-weighted training loss in rmse - train: 0.04/ valid: 0.08
Sample-weighted training loss in rmse - train: 0.15/ valid: 0.34
I could manage to replicate the rmse result in Non-weighted data simply using model.pred() and then calculate by myself.
But I don't know how to calculate the Sample-weighted one.

If the calculation is not too complicated, can u guys share how to calculate the loss with weights?

Thanks

catboost version: 0.24
Operating System: Linux Ubuntu 20.04
CPU:
GPU: 2080ti

**Comment by DeanLeeFumu** [comment]
I've found the formula in below issue:
https://github.com/catboost/catboost/issues/150

np.sqrt(np.sum(weights * ((pred - test_labels) ** 2)) / np.sum(weights)))
